libs/wallet/src/web3-react/Web3Provider/hooks/useEagerlyConnect.ts

```diff
@@ -24,19 +28,33 @@ async function connect(connector: Connector) {
 }
 
 export function useEagerlyConnect(selectedWallet: ConnectionType | undefined, selectedWalletBackfilled: boolean) {
+  const selectedEip6963ProviderInfo = useSelectedEip6963ProviderInfo()
+  const setEip6963Provider = useSetEip6963Provider()
+
   useEffect(() => {
-    if (isInjectedWidget()) {
-      connect(injectedWidgetConnection.connector)
+    const isIframe = window.top !== window.self
+
+    if (isInjectedWidget() || (isIframe && getIsInjected())) {
+      connect(injectedWalletConnection.connector)
     }
 
     // Try to connect to Gnosis Safe only when the app is opened in an iframe
-    if (window.top !== window.self) {
+    if (isIframe) {
       connect(gnosisSafeConnection.connector)
     }
 
     connect(networkConnection.connector)
 
     if (selectedWallet) {
+      const connection = getWeb3ReactConnection(selectedWallet)
+      const cachedProviderRdns = jotaiStore.get(selectedEip6963ProviderRdnsAtom)
```

Comment:
I could use `useAtomValue(selectedEip6963ProviderRdnsAtom)` instead, but for some reason it doesn't work as expected. With `useAtomValue` it doesn't have enough time to fullfill data from `localStorage`.
Anyway, this `useEffect()` intentionally doesn't have dependencies, so it's even better  if we use `jotaiStore.get()` here.